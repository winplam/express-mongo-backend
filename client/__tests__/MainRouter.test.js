import React from 'react'
import { Router } from 'react-router-dom'
import { getByRole, render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { createMemoryHistory } from 'history'
import Home from '../core/Home'
import MainRouter from '../MainRouter'

test('home container matches snapshot', () => {
  const { container } = render(<Home/>)
  expect(container).toMatchSnapshot()
})

const renderWithRouter = (component) => {
  const history = createMemoryHistory()
  return {
    ...render(
      <Router history={history}>
        {component}
      </Router>
    )
  }
}
it('should render the home page', () => {
  const { container, getByText } = renderWithRouter(<MainRouter/>)
  expect(container.innerHTML).toMatch(/home page/i)
  expect(getByText('Home Page')).toBeDefined()
  // ToDo: Test click away from and then click back to home page using home page icon button
  userEvent.click(getByText(/sign in/i))
  expect(screen.queryByText(/home page/i)).toBeNull()
  // userEvent.click(getByText(''))
  // expect(getByText('Home Page')).toBeDefined()
})

it('should navigate to the users page', () => {
  const { container, getByText } = renderWithRouter(<MainRouter/>)
  expect(container.innerHTML).toMatch('Users')
  // ToDo: Test display of All Users page
  // userEvent.click(getByText(/users/i))
  // expect(container.innerHTML).toMatch('All Users')
})

it('should navigate to the sign up page', () => {
  const { container, getByText } = renderWithRouter(<MainRouter/>)
  expect(container.innerHTML).toMatch(/home page/i)
  userEvent.click(getByText(/sign up/i))
  expect(container.innerHTML).toMatch(/sign up/i)
})

it('should navigate to the sign in page', () => {
  const { container, getByText } = renderWithRouter(<MainRouter/>)
  expect(container.innerHTML).toMatch(/home page/i)
  userEvent.click(getByText(/sign in/i))
  expect(container.innerHTML).toMatch(/sign in/i)
})
