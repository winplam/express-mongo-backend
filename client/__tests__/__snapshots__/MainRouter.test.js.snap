// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home container matches snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiCard-root makeStyles-card-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <h6
      class="MuiTypography-root makeStyles-title-2 MuiTypography-h6"
    >
      Home Page
    </h6>
    <p
      class="MuiTypography-root makeStyles-credit-4 MuiTypography-body2 MuiTypography-colorTextSecondary"
    >
      Photo by
      <a
        href="https://unsplash.com/@boudewijn_huysmans"
        rel="noopener noreferrer"
        target="_blank"
      >
        Boudewijn Huysmans
      </a>
      on Unsplash
    </p>
    <div
      class="MuiCardContent-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        Welcome to the MERN Skeleton home page.
      </p>
    </div>
  </div>
</div>
`;
